[ADH-3712] Added PostgreSQL support in metastore

tigrulya-exe opened the pull request:
1. Adapted migration scripts to support all DBMS;
2. Refactored `InotifyEventApplier` and moved the SQL scripts generation logic from it to the relevant DAOs;
3. Moved database metadata fetch logic from `MetaStoreUtils` to the DBMS specific `DbMetadataProvider` interface implementations;
4. Added several Postgres specific DAO implementations;
5. Refactored the metastore-related tests and parameterized them by `DbType`. MySQL and PostgreSQL instances are started using Testcontainers.

tigrulya-exe committed:
[ADH-3712] Added PostgreSQL support in metastore

xardazzzzzz commented on pom.xml:
```diff
@@ -219,6 +220,11 @@
+      <dependency>
+        <groupId>org.postgresql</groupId>
+        <artifactId>postgresql</artifactId>
+        <version>42.6.0</version>
```
move version to properties

tigrulya-exe replied:
done

tigrulya-exe commented:
Removed `hadoop-hdfs-client` dependency redefinition in `smart-metastore` in last change, because it breaks build of the project with enabled `hadoop-2.7` profile. Replaced with direct test dependency `jackson-annotations` to avoid the dependency version problems during tests using testcontainers.
